src/main/java/com/metamx/emitter/core/Emitters.java

```diff
@@ -43,21 +48,39 @@ public static Emitter create(Properties props, HttpClient httpClient, Lifecycle
   public static Emitter create(Properties props, HttpClient httpClient, ObjectMapper jsonMapper, Lifecycle lifecycle)
   {
     Map<String, Object> jsonified = Maps.newHashMap();
-
+    Class<? extends EmitterFactory> emitterFactoryClass;
     if (props.getProperty(LOG_EMITTER_PROP) != null) {
       jsonified.put("logging", makeLoggingMap(props));
+      emitterFactoryClass = LoggingEmitterFactory.class;
     }
     else if (props.getProperty(HTTP_EMITTER_PROP) != null) {
       jsonified.put("http", makeHttpMap(props));
+      emitterFactoryClass = HttpEmitterFactory.class;
     }
     else if (props.getProperty(PARAMETRIZED_HTTP_EMITTER_PROP) != null) {
       jsonified.put("parametrized", makeParametrizedHttpMap(props));
+      emitterFactoryClass = ParametrizedUriEmitterFactory.class;
+    }
+    else if (props.getProperty(CUSTOM_EMITTER_FACTORY_PROP) !=null) {
+      jsonified = makeCustomFactoryMap(props);
+      try {
+        emitterFactoryClass = (Class<? extends EmitterFactory>) Class.forName(props.getProperty(CUSTOM_EMITTER_FACTORY_PROP));
+      }
+      catch (ClassNotFoundException e) {
+        throw new ISE(e, "Invalid class name set for [%s]", CUSTOM_EMITTER_FACTORY_PROP);
+      }
     }
     else {
-      throw new ISE("Unknown type of emitter. Please set [%s] or [%s]", LOG_EMITTER_PROP, HTTP_EMITTER_PROP);
+      throw new ISE(
+          "Unknown type of emitter. Please set [%s], [%s], [%s] or provide class implementing com.metamx.emitter.core.factory.EmitterFactory via [%s]",
+          LOG_EMITTER_PROP,
+          HTTP_EMITTER_PROP,
+          PARAMETRIZED_HTTP_EMITTER_PROP,
+          CUSTOM_EMITTER_FACTORY_PROP
+      );
     }
 
-    return jsonMapper.convertValue(jsonified, EmitterBuilder.class).build(jsonMapper, httpClient, lifecycle);
+    return jsonMapper.convertValue(jsonified, emitterFactoryClass).build(jsonMapper, httpClient, lifecycle);
```

Comment:
Actually, can you get it to where `jsonMapper` is not needed in the builder by using such a construction method?